packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -24,6 +24,19 @@ const double _kFloatingActionButtonMargin = 16.0; // TODO(hmuller): should be de
 const Duration _kFloatingActionButtonSegue = const Duration(milliseconds: 200);
 final Tween<double> _kFloatingActionButtonTurnTween = new Tween<double>(begin: -0.125, end: 0.0);
 
+/// Returns a path for a notch in the outline of a shape.
+///
+/// The path makes a notch in the host shape that can contain the guest shape.
+///
+/// The [host] is the bounding rectangle for the host shape, and [guest] is the
+/// bounding rectangle for the guest shape.
```

Comment:
Let's say something like "The [host] is the bounding rectangle for the shape into which the notch is to be drawn (for example, a [BottomAppBar]), and the [guest] is the bounding rectangle for the widget that encroaches upon the host (for example, a [FloatingActionButton])."

Also, use backticks instead of square brackets for arguments.

Reply:
partially adopted your description.
I didn't mentioned BAB/FAB as the BAB does not exist yet, I added a TODO to update the doc here with an example once BAB lands.